### DISCUSSION TITLE
Volume automatically returns to 0 at the start of connection

### DISCUSSION THREAD

**Discussion by thanhtung386** [pschatzmann/ESP32-A2DP#391] [Q&A] [answered]
Hi guys,
I am having a problem on ESP32-A2DP. The volume automatically returns to 0 at the start of connection. 
This is may code
#include "BluetoothA2DPSink.h"
BluetoothA2DPSink a2dp_sink;
void setup() {
  //Internal DAC config
  const i2s_config_t i2s_config = {
      .mode = (i2s_mode_t) (I2S_MODE_MASTER | I2S_MODE_TX | I2S_MODE_DAC_BUILT_IN),
      .sample_rate = 44100, // corrected by info from bluetooth
      .bits_per_sample = (i2s_bits_per_sample_t) 16, /* the DAC module will only take the 8bits from MSB */
      .channel_format =  I2S_CHANNEL_FMT_RIGHT_LEFT,
      .communication_format = (i2s_comm_format_t)I2S_COMM_FORMAT_STAND_MSB,
      .intr_alloc_flags = 0, // default interrupt priority
      .dma_buf_count = 8,
      .dma_buf_len = 64,
      .use_apll = false,
      .tx_desc_auto_clear = false
  };
  a2dp_sink.set_i2s_config(i2s_config); 
  a2dp_sink.start("SPEAKER"); 
}

void loop() {
}

**Comment by pschatzmann** [comment, marked as answer]
The volume is controlled by the volume on your phone!
